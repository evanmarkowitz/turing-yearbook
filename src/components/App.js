import React, { Component } from 'react';
import Cohort from './Cohort';
import people from '../data/yearbook-data.js';
import './App.css';
import AddStudent from './AddStudent'
import { BrowserRouter as Router, Route, Link } from "react-router-dom";


class App extends Component {
  constructor() {
    super();
    this.state = {
      staff: people.staff,
      students: people.students,
      isHidden: true,

    }
  }
  addStudent = newStudent => {
    this.setState({students: [...this.state.students, newStudent]})
  }

  toggleIsHidden = () => {
    this.setState({isHidden: !this.state.isHidden})
  }


  render() {
    const Staff = () => (
      <div>
        <h2>Staff</h2>
        <Cohort data={this.state.staff} />
      </div>
    );
    
   const Students = () => (
      <div>
        <h2>Students</h2>
        <Cohort data={this.state.students} />
      </div>
    );
    
    return (
      <div className="App">
        <header className="App-header">
         <h1>Turing Yearbook</h1>
        </header>
        <button button onClick={this.toggleIsHidden}>Add Person</button>
        {!this.state.isHidden && <AddStudent addStudent={this.addStudent}/>}
        <Router>
          <Link to='' component={Staff}/>
          <Link to="/staff" className="togglePerson"><span>Staff</span></Link> | 
          <Link to="/students"><span>Students</span></Link>
          <Route path="/staff" component={Staff} />
          <Route path="/students" component={Students} />
      </Router>
      </div>
    );
  }
}

export default App;
