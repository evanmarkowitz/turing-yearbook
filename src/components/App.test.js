import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Cohort', () => { 

  it('should match the snapshot with all data passed in correctly', () => { 
    const wrapper = mount( <App
      />);
    expect(wrapper).toMatchSnapshot(); 
  }); 
  
}); 