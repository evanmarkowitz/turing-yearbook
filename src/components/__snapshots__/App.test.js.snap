// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot with all data passed in correctly 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <h1>
        Turing Yearbook
      </h1>
    </header>
    <button
      button={true}
      onClick={[Function]}
    >
      Add Person
    </button>
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Link
          component={[Function]}
          to=""
        >
          <a
            component={[Function]}
            href="/"
            onClick={[Function]}
          />
        </Link>
        <Link
          className="togglePerson"
          to="/staff"
        >
          <a
            className="togglePerson"
            href="/staff"
            onClick={[Function]}
          >
            <span>
              Staff
            </span>
          </a>
        </Link>
         |
        <Link
          to="/students"
        >
          <a
            href="/students"
            onClick={[Function]}
          >
            <span>
              Students
            </span>
          </a>
        </Link>
        <Route
          component={[Function]}
          path="/staff"
        />
        <Route
          component={[Function]}
          path="/students"
        />
      </Router>
    </BrowserRouter>
  </div>
</App>
`;
